### PR TITLE
Continue repository setup

### PR DESCRIPTION
**Pull request by alindima**
- Add `.gitignore` file
- Modified `Cargo.toml`
- Committed `Cargo.lock`
- Add BSD-3-Clause License
- Small Readme Update

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.github
+/target
+.vscode
```
Comment (andreeaflorescu) [inline .gitignore:3]: This should be setup at the host level. Otherwise you're gonna end up with lots of IDE specific ignore entries.

There are some tips here about how to do this: http://egorsmirnov.me/2015/05/04/global-gitignore-file.html

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.github
```
Comment (andreeaflorescu) [inline .gitignore:1]: Why is this added?

File: Cargo.lock [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# This file is automatically @generated by Cargo.
```
Comment (andreeaflorescu) [inline Cargo.lock:1]: We might not need to commit this depending on the decision to include the binary in this repository. I would leave it out for now.

File: .gitignore (hunk 1 above)
Comment (alindima) [reply .gitignore:3]: good idea, thanks

File: .gitignore (hunk 2 above)
Comment (alindima) [reply .gitignore:1]: I did this by mistake

File: Cargo.lock (hunk 3 above)
Comment (alindima) [reply Cargo.lock:1]: good idea

File: .gitignore (hunk 2 above)
Comment (alindima) [reply .gitignore]: removed it

File: Cargo.lock (hunk 3 above)
Comment (alindima) [reply Cargo.lock]: removed

**Review by andreeaflorescu — DISMISSED** [review]

**Review by jiangliu — APPROVED** [review]

**Comment by alindima** [comment]
@jiangliu @andreeaflorescu I had to fix some commit style errors, all green now

**Review by andreeaflorescu — APPROVED** [review]

**Review by alexandruag — APPROVED** [review]
